crates/ide_completion/src/completions/record.rs

```diff
@@ -59,10 +59,13 @@ pub(crate) fn complete_record_literal(
     }
 
     if let hir::Adt::Struct(strukt) = ctx.expected_type.as_ref()?.as_adt()? {
-        let module =
-            if let Some(module) = ctx.scope.module() { module } else { strukt.module(ctx.db) };
-
-        let path = module.find_use_path(ctx.db, hir::ModuleDef::from(strukt));
+        let path = if ctx.path_qual().is_none() {
+            ctx.scope
+                .module()
+                .and_then(|module| module.find_use_path(ctx.db, hir::ModuleDef::from(strukt)))
+        } else {
+            None
+        };
```

Comment:
This won't quite work, I imagine in the following scenario we get a completion for the struct still though with no path, even though we shouldn't show it at all.
```rs
mod x { pub struct Bar; }

fn main() {
    let _: x::Bar = foo::$0;
}
```
So I think we should either not show this completion at all in qualified paths, or make sure the path resolves to the struct we want to show a literal completion for